exasol_transformers_extension/udfs/models/filling_mask_udf.py

```diff
@@ -134,65 +124,63 @@ def get_prediction(self, model_df: pd.DataFrame) -> pd.DataFrame:
         :return: The dataframe where the model_df is formatted with the
         prediction results
         """
-        preds, labels = self._predict_model(model_df)
-        pred_df = self._prepare_prediction_dataframe(model_df, preds, labels)
+        pred_df_list = self._predict_model(model_df)
+        pred_df = self._prepare_prediction_dataframe(model_df, pred_df_list)
         return pred_df
 
-    def _predict_model(self, model_df: pd.DataFrame) -> \
-            Tuple[List[float], List[str]]:
+    def _predict_model(self, model_df: pd.DataFrame) -> List[pd.DataFrame]:
         """
         Predict the given text list using recently loaded models, return
         probability scores and filled texts
 
         :param model_df: The dataframe to be predicted
 
-        :return: A tuple containing prediction score list and filled texts list
+        :return: List of dataframe includes prediction details
         """
+        top_k = int(model_df['top_k'].iloc[0])
         text_data_raw = list(model_df['text_data'])
-        text_data_with_valid_mask_token = \
-            [text_data.replace(self.mask_token,
-                               self.last_created_pipeline.tokenizer.mask_token)
-             for text_data in text_data_raw]
-        results = self.last_created_pipeline(text_data_with_valid_mask_token)
+        text_data_with_valid_mask_token = [
+            text_data.replace(
+                self.mask_token,
+                self.last_created_pipeline.tokenizer.mask_token
+            ) for text_data in text_data_raw]
+        results = self.last_created_pipeline(
+            text_data_with_valid_mask_token, top_k=top_k)
 
         #  Batch prediction returns list of list while single prediction just
-        #  return a list. In case of batch predictions, we need to flatten
-        #  2D prediction results to 1D list
-        results = sum(results, []) if type(results[0]) == list else results
+        #  return a list. In order to ease dataframe operations, convert single
+        #  prediction to list of list.
+        results = [results] if len(text_data_raw) == 1 else results
 
-        filled_texts = []
-        scores = []
+        columns = ["sequence", "score"]
+        results_df_list = []
         for result in results:
-            filled_texts.append(result['sequence'])
-            scores.append(result['score'])
-
-        return scores, filled_texts
+            result_df = pd.DataFrame(result)
+            result_df = result_df[columns].rename(
+                columns={"sequence": "filled_text"})
+            results_df_list.append(result_df)
```

Comment:
extract to method

Reply:
These refactorings are actually part of the refactoring ticket #19  . All model udfs have these methods, somehow we collect them in a common class

Reply:
ok, agree